### PR TITLE
Draft changes for itemHandler

### LINKED ISSUE #683: [BUG] Entities not showing up properly in schematic previews.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Version?

- [X] I am running the latest **beta** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [X] Single Player
- [X] Multi Player

### Minecraft Version

1.20.1

### MineColonies Version

1.20.1-1.1.574-BETA

### Structurize Version

1.20.1-1.0.733-RELEASE

### Related Mods and their Versions

_No response_

### Current Behavior

When previewing schematics, the entities, such as item frames, armor stands, paintings. Do not always show up properly, but it builds just fine. (Taken on Cave Colonies)

When testing on the most recent minecolonies/structurize, they did show up, but at random. Possibly a distance issue.

https://github.com/ldtteam/minecolonies/assets/85356635/9aa59794-ee52-4d75-9de0-b45fbaf85ef5

### Expected Behavior

Being able to properly see the entities, so players can see what actual costs of building will be. Otherwise there may be some nasty surprises.

### Reproduction Steps

Preview a schematic with entities in it. The example I showed is Aquatica: JinoHotel level 5.

### Logs

None

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by Nightenom**
# Changes proposed in this pull request
- requires ldtteam/blockui#101
- reworks how we handle itemhandlers etc., should produce much more precise content handling, including nested inventories like shulkers
- added "show resources" for build tool, scan tool doesnt do content extraction, build tool runs same methods as structure placer, so technically builder should show exactly the same as build tool resources
- tested in single with blueprint shown on picture - all vanilla BEs with content, also tested with most entities

## Testing
- [x] Yes I tested this before submitting it.
- [ ] I also did a multiplayer test.

Review please

image
image

### CODE REVIEW

File: src/main/java/com/ldtteam/structurize/api/ItemStackUtils.java [hunk 1]
```diff
@@ -81,24 +82,7 @@ public static List<ItemStack> getItemStacksOfTileEntity(final CompoundTag compou
...
-        for (int i = 0; i < listtag.size(); ++i)
-        {
-            final CompoundTag compoundtag = listtag.getCompound(i);
-            final ItemStack stack = ItemStack.parseOptional(provider, compoundtag);
-            if (!stack.isEmpty())
-            {
-                items.add(stack);
-            }
-        }
+        SingleBlockFakeLevelGetter.unset(itemHandlerFakeLevel, tileEntity);
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/structurize/api/ItemStackUtils.java]: prepare + unset seems kinda unstable, isnt there any single shot method we could use?

**Comment by Nightenom** [comment]
@Raycoms this might probably do, but i havent tested it properly, also i have done nothing towards to double chest and alike

Also this should be accompanied by changed during placement/pre-placement traversals do we require exactly once the content of BE

File: src/main/java/com/ldtteam/structurize/util/BlockUtils.java [hunk 2]
```diff
@@ -264,6 +265,7 @@ public static boolean isWater(final BlockState iBlockState)
         return iBlockState.getBlock() == Blocks.WATER;
     }
 
+    @Deprecated(forRemoval = true, since = "1.21")
     private static Item getItem(final BlockState blockState)
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/structurize/util/BlockUtils.java:269]: we will still need solutions for this kind of things, don't just deprecate it =D

**Review by Raycoms — DISMISSED** [review]
Should otherwise be okay, was did tested?
(Please don't deprecate the others for now)

File: src/main/java/com/ldtteam/structurize/util/BlockUtils.java (hunk 2 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/structurize/util/BlockUtils.java:269]: It's in blockui for two version already

File: src/main/java/com/ldtteam/structurize/client/gui/WindowScan.java [hunk 3]
```diff
@@ -478,7 +478,7 @@ private void updateResources()
                 boolean handled = false;
                 for (final IPlacementHandler handler : PlacementHandlers.handlers)
                 {
-                    if (handler.canHandle(world, BlockPos.ZERO, blockState))
```
Comment (Nightenom) [inline src/main/java/com/ldtteam/structurize/client/gui/WindowScan.java:481]: This was weird, especially no need to load world at zero during compute resources step

**Pull request merged**